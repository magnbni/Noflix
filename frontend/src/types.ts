export type MovieEdge = {
  cursor: string;
  node: MovieType;
};

export type UserEdge = {
  cursor: string;
  node: UserType;
};

export interface MovieType {
  Id: string;
  title: string;
  releaseDate: string;
  overview: string;
  voteAverage: number;
  posterPath: string;
}

export interface UserType {
  email: string;
  password: string;
}

export interface GenreEdge {
  node: GenreType;
}

export interface GenreType {
  name: string;
  id: string;
}

export interface RateProps {
  initValue: number | null;
  handleUserRating: (rating: number) => void;
  movieId: string;
  open: boolean;
}
