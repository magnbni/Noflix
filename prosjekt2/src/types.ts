export type MovieEdge = {
  cursor: string;
  node: MovieType;
};

export interface MovieType {
  Id: string;
  title: string;
  releaseDate: string;
  overview: string;
  voteAverage: number;
  posterPath: string;
}

export interface GenreEdge {
  node: GenreType;
}

export interface GenreType {
  name: string;
  id: string;
}

export interface RateProps {
  initValue: number | null;
  handleUserRating: (rating: number) => void;
  movieId: string;
  open: boolean;
}
