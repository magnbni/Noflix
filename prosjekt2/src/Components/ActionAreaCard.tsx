import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import { CardActionArea, Skeleton } from "@mui/material";
import { ReadOnlyRating } from "./BasicRating";
import { MovieType } from "../types";

/*
  Custom action card used to present each individual movie.
*/

export interface ActionAreaCardProps {
  movie: MovieType | undefined;
}

export default function ActionAreaCard({ movie }: ActionAreaCardProps) {
  const truncatedTitle =
    movie && movie.title
      ? movie.title.length > 19
        ? movie.title.slice(0, 19) + "..."
        : movie.title
      : "";

  const date =
    movie && movie.releaseDate ? (
      movie.releaseDate.length >= 9 ? (
        new Date(movie.releaseDate).toLocaleDateString("en-GB")
      ) : (
        <br />
      )
    ) : (
      ""
    );

  return (
    // Material-UI Card component with custom styles
    <Card sx={{ width: 300, heigth: 350 }} data-testid="ActionAreaCard">
      <CardActionArea>
        <br />
        {movie && (
          <div
            data-testid="movie-poster"
            style={{
              margin: "auto",
              width: "240px",
              height: "350px",
              backgroundSize: "cover",
              backgroundPosition: "center",
              backgroundImage:
                "url(https://image.tmdb.org/t/p/original//" +
                movie.posterPath +
                ")",
              borderRadius: "4px",
              border: "2px solid #000000",
            }}
          />
        )}
        {!movie && (
          <Skeleton
            data-testid="skeleton-loader"
            variant="rectangular"
            width={240}
            height={350}
            style={{
              margin: "auto",
              borderRadius: "4px",
              border: "2px solid #000000",
            }}
          />
        )}
        <CardContent>
          <Typography gutterBottom variant="h5" component="div">
            {truncatedTitle === "" ? <Skeleton /> : truncatedTitle}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            {date === "" ? <Skeleton /> : date}
          </Typography>
        </CardContent>
        {movie && (
          <div style={{ paddingLeft: "13px", paddingBottom: "10px" }}>
            {movie.voteAverage >= 1 && movie.voteAverage <= 10
              ? ReadOnlyRating(movie.voteAverage / 2)
              : ReadOnlyRating(null)}
          </div>
        )}
        {!movie && (
          <div style={{ paddingLeft: "13px", paddingBottom: "10px" }}>
            <Skeleton variant="rectangular" width={100} height={27} />
          </div>
        )}
      </CardActionArea>
    </Card>
  );
}
