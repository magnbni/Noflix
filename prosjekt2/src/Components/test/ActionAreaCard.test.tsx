import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import ActionAreaCard from "../ActionAreaCard";
import { MovieType } from "../../types";

describe("ActionAreaCard", () => {
  test("Snapshot test of page", () => {
    const movie: MovieType = {
      Id: "MOCK",
      title: "MOCK",
      releaseDate: "MOCK",
      overview: "MOCK",
      voteAverage: 0,
      posterPath: "MOCK",
    };
    const page = render(<ActionAreaCard {...movie} />);
    expect(page).toMatchSnapshot();
  });
});
