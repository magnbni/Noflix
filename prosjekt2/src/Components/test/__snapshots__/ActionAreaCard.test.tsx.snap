// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionAreaCard > Snapshot test of page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-196akmx-MuiPaper-root-MuiCard-root"
        data-testid="ActionAreaCard"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <br />
          <div
            data-testid="movie-poster"
            style="margin: auto; width: 240px; height: 350px; background-size: cover; background-position: center; background-image: url(https://image.tmdb.org/t/p/original///example.jpg); border-radius: 4px; border: 2px solid #000000;"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              Example Movie Title
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              01/01/2023
            </p>
          </div>
          <div
            style="padding-left: 13px; padding-bottom: 10px;"
          >
            <div
              class="MuiBox-root css-0"
            >
              <span
                aria-label="4 Stars"
                class="MuiRating-root MuiRating-sizeMedium Mui-readOnly MuiRating-readOnly css-ryrseu-MuiRating-root"
                role="img"
              >
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-196akmx-MuiPaper-root-MuiCard-root"
      data-testid="ActionAreaCard"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <br />
        <div
          data-testid="movie-poster"
          style="margin: auto; width: 240px; height: 350px; background-size: cover; background-position: center; background-image: url(https://image.tmdb.org/t/p/original///example.jpg); border-radius: 4px; border: 2px solid #000000;"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
          >
            Example Movie Title
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            01/01/2023
          </p>
        </div>
        <div
          style="padding-left: 13px; padding-bottom: 10px;"
        >
          <div
            class="MuiBox-root css-0"
          >
            <span
              aria-label="4 Stars"
              class="MuiRating-root MuiRating-sizeMedium Mui-readOnly MuiRating-readOnly css-ryrseu-MuiRating-root"
              role="img"
            >
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarBorderIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
